### PR TITLE
RBAC

### PR DESCRIPTION
**Pull request by justnero**

### CODE REVIEW

File: openapi/components/schemas/Role.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    type: array
+    description: >-
+      Any embedded objects available that are requested by the `expand`
+      querystring parameter.
+    readOnly: true
+    minItems: 1
+    items:
+      anyOf:
+        - $ref: ./Embeds/SeniorRolesEmbed.yaml
+        - $ref: ./Embeds/JuniorRolesEmbed.yaml
```
Comment (adamaltman) [inline openapi/components/schemas/Role.yaml:61]: This is recursive... senior contains junior... junior contains senior, etc... Not sure it's a good idea.

Comment (justnero) [reply openapi/components/schemas/Role.yaml:61]: We can either remove senior here or just limit the depth

File: openapi/components/schemas/Role.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  _links:
+    type: array
+    description: The links related to a resource.
+    readOnly: true
+    minItems: 1
+    items:
+      anyOf:
+        - $ref: ./Links/SelfLink.yaml
+        - $ref: ./Links/SeniorRolesLink.yaml
+        - $ref: ./Links/JuniorRolesLink.yaml
```
Comment (adamaltman) [inline openapi/components/schemas/Role.yaml:50]: this kind of confuses me... what do these objects actually look like?

Comment (justnero) [reply openapi/components/schemas/Role.yaml:50]:
```json
[
	{
		"rel": "seniorRoles",
		"href": "https://api.rebilly.com/roles?filter=juniorIds:foo"
	},
	{
		"rel": "juniorRoles",
		"href": "https://api.rebilly.com/roles?filter=seniorIds:foo"
	}
]
```

File: openapi/users.yaml [hunk 3]
```diff
@@ -189,6 +189,10 @@ tags:
     description: |
       Membership describe the relationships between organizations and users.
       For example, one user could have more than one organization.
+  - name: Roles
+    description: |
+      Role is an implementation of general hierarcy RBAC. Senior role inherits all of it's junior ACLs plus it's own ACL.
```
Comment (adamaltman) [inline openapi/users.yaml:194]:
```suggestion
      Roles are an implementation of the limited hierarchical RBAC. A senior role inherits all of it's juniors' ACLs plus its own ACL.
```

**Comment by justnero** [comment]
@Rebilly/php please review

**Comment by justnero** [comment]
@Rebilly/php please review

**Comment by justnero** [comment]
@Rebilly/js please review

**Comment by justnero** [comment]
@Rebilly/php please review

**Comment by ashkarpetin** [comment]
@justnero Operations are missing required x-sdk-operation-name property. Please fix it.

**Comment by justnero** [comment]
@ashkarpetin fixed, thanks!

**Review by ashkarpetin — APPROVED** [review]

**Review by iamdevice — APPROVED** [review]

**Pull request merged**